File: src/completed/java/euler0015.java
/**
 * Created by Holden on 12/02/2016. - COMPLETED
 *
 * Problem 15 - Lattice Paths
 *
 */

public class Prob0015 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        long answer = latticePaths(20);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static long latticePaths(int dimension) {
        return EulerUtil.combine(2*dimension, dimension);
    }
}

File: src/completed/python/euler0015.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 4/16/2016
#
# Problem 15 - Lattice Paths
#
###

from math import factorial as f

def lattice_paths(grid_size):
    choose = grid_size + (grid_size + 1) - 1
    return f(choose) // (f(grid_size) * f(choose - grid_size))